### PR TITLE
Added basic test for ACE on RKE2

### PR DESCRIPTION
**Pull request by MSpencer87**
Added basic tests for provisioning RKE2 clusters with Local Authorized Cluster Endpoint enabled

### CODE REVIEW

File: tests/v2/validation/provisioning/provisioning_rke2_ace_test.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+			"worker":       true,
+		},
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles []map[string]bool
+		client    *rancher.Client
+	}{
+		{"Single Control Plane", nodeRoles0, r.client},
```
[the remark this answers is not in the document]
Comment (igomez06) [reply tests/v2/validation/provisioning/provisioning_rke2_ace_test.go:130]: I see what you are saying @izaac , it is possible to just run one at a time. When both are ran, you will see which one fails, but it is possible to run them one at a time depending on the command. Does that satisfy what you are looking for?

File: tests/v2/validation/provisioning/provisioning_rke2_ace_test.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+					require.NoError(r.T(), err)
+
+					originalResp, err := original.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+					require.NoError(r.T(), err)
+
+					for _, pod := range originalResp.Items {
+						r.T().Logf("Pod %v", pod.GetName())
+					}
+
+					contexts, err := kubeconfig.GetContexts(kubeConfig)
```
Comment (igomez06) [inline tests/v2/validation/provisioning/provisioning_rke2_ace_test.go:193]: If you don't need to the original context switch, then this call is only needed once. Does that make sense?

File: tests/v2/validation/provisioning/provisioning_rke2_ace_test.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+					require.NoError(r.T(), err)
+
+					original, err := r.client.SwitchContext(clusterResp.Name, kubeConfig)
+					require.NoError(r.T(), err)
+
+					originalResp, err := original.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+					require.NoError(r.T(), err)
+
+					for _, pod := range originalResp.Items {
+						r.T().Logf("Pod %v", pod.GetName())
```
Comment (igomez06) [inline tests/v2/validation/provisioning/provisioning_rke2_ace_test.go:190]: @izaac do we want to log all the pod names?

File: tests/v2/validation/provisioning/setup-common.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package provisioning
```
Comment (igomez06) [inline tests/v2/validation/provisioning/setup-common.go:1]: rename this file `setup.go` let's not name anything with common, and avoid using underscores except for test files.

File: tests/v2/validation/provisioning/provisioning_rke2_ace_test.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RKE2ACETestSuite struct {
+	suite.Suite
+	client                   *rancher.Client
+	session                  *session.Session
```
[the remark this answers is not in the document]
Comment (izaac) [reply tests/v2/validation/provisioning/provisioning_rke2_ace_test.go:29]: We should do it as standard user, good catch @igomez06.

Also I think you answer my question too, if there's a way to run it individually it's ok then.

File: tests/v2/validation/provisioning/provisioning_rke2_ace_test.go (hunk 3 above)
Comment (izaac) [reply tests/v2/validation/provisioning/provisioning_rke2_ace_test.go:190]: I don't think we need all the pods if we're sure context-switching worked and the user is able to request information from the cluster using kubectl that will be enough.

File: tests/v2/validation/provisioning/provisioning_rke2_ace_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (izaac) [reply tests/v2/validation/provisioning/provisioning_rke2_ace_test.go:130]: @igomez06  If there's a way Im good then, it's probably just me not knowing all the framework execution options.

File: tests/v2/validation/provisioning/provisioning_rke2_ace_test.go [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+					clusterInfo, err := testSessionClient.Provisioning.Clusters(namespace).Get(context.TODO(), clusterResp.Name, metav1.GetOptions{})
+					require.NoError(r.T(), err)
+
+					clusterID := clusterInfo.Status.ClusterName
+
+					kubeConfig, err := kubeconfig.GetKubeconfig(r.client, clusterID)
+					require.NoError(r.T(), err)
+
+					original, err := r.client.SwitchContext(clusterResp.Name, kubeConfig)
+					require.NoError(r.T(), err)
```
[the remark this answers is not in the document]
Comment (MSpencer87) [reply tests/v2/validation/provisioning/provisioning_rke2_ace_test.go:184]: You're correct, the original context is default

**Comment by izaac** [comment]
I reviewed the Jenkins execution and seems that the failures are related to https://github.com/rancher/rke2/issues/2480

We just need the code changes to be addressed for a second review pass

**Comment by bmdepesa** [comment]
@MSpencer87 what's the current state of this?

**Comment by MSpencer87** [comment]
Replaced by 

[v2.6] https://github.com/rancher/rancher/pull/40276

and

[v2.7] https://github.com/rancher/rancher/pull/40273

**Pull request closed without merging**